client/service/test/src/lib.rs

```diff
@@ -304,7 +302,6 @@ impl<G, E, F, L, U> TestNet<G, E, F, L, U> where
 			let addr = node_config.network.listen_addresses.iter().next().unwrap().clone();
 			let service = SyncService::from(light(node_config).expect("Error creating test node service"));
 
-			executor.spawn(service.clone().map_err(|_| ()));
```

Comment:
tbh I'm not sure why those tasks are spawn because the Future impl Service is just waiting indefinitely and stops if there is an essential task failing. But since this is spawn like a background process, nobody is doing anything with the result anyway. Will see if it breaks on the CI

Reply:
The `Service` used to properly own the tasks that it runs, and process them manually if no executor is available, rather than always spawning these tasks in the background.

Reply:
That explains everything